etl/steps/data/garden/ember/2022-12-13/yearly_electricity.py

```diff
@@ -0,0 +1,387 @@
+        Data in long format for all categories, after harmonizing country names.
+
+    Returns
+    -------
+    table : catalog.Table
+        Table of processed data for the given category.
+
+    """
+    # Prepare wide table.
+    table = prepare_wide_table(df=df, category="Electricity demand")
+
+    # Add population to data
+    table = add_population(df=table, warn_on_missing_countries=False)
+
+    # Recalculate demand per capita.
+    # We could do this only for region aggregates (since they do not have per capita values),
+    # but we do this for all countries, to ensure per-capita variables are consistent with our population data.
+    table["Demand per capita (kWh)"] = (
+        pd.DataFrame(table)["Demand (TWh)"] * TWH_TO_KWH / pd.DataFrame(table)["population"]
+    )
+
+    # Delete the original demand per capita column.
+    table = table.drop(columns="Demand per capita (MWh)")
+
+    return table
+
+
+def process_power_sector_emissions(df: pd.DataFrame) -> catalog.Table:
+    """Create table with processed data of category "Power sector emissions".
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Data in long format for all categories, after harmonizing country names.
+
+    Returns
+    -------
+    table : catalog.Table
+        Table of processed data for the given category.
+
+    """
+    # Prepare wide table of emissions data.
+    table = prepare_wide_table(df=df, category="Power sector emissions")
+
+    # Add carbon intensity.
+    # In principle this only needs to be done for region aggregates, but we do it for all countries and check that
+    # the results are consistent with the original data.
+    # Prepare wide table also for electricity generation (required to calculate carbon intensity).
+    electricity = prepare_wide_table(df=df, category="Electricity generation")[
+        ["country", "year", "Total Generation (TWh)"]
+    ]
+    # Add total electricity generation to emissions table.
+    table = pd.merge(table, electricity, on=["country", "year"], how="left")
+    # Rename the original carbon intensity column as a temporary column called "check".
+    intensity_col = "CO2 intensity (gCO2/kWh)"
+    table = table.rename(columns={intensity_col: "check"})
+    # Calculate carbon intensity for all countries and regions.
+    table[intensity_col] = (
+        pd.DataFrame(table)["Total emissions (mtCO2)"] * MT_TO_G / (table["Total Generation (TWh)"] * TWH_TO_KWH)
+    )
+
+    # Check that the new carbon intensities agree (within 1 % of mean average percentage error) with the original
+    # ones (where carbon intensity was given, namely for countries, not aggregate regions).
+    mape = 100 * abs(table.dropna(subset="check")[intensity_col] - table["check"].dropna()) / table["check"].dropna()
+    assert mape.max() < 1, "Calculated carbon intensities differ from original ones by more than 1 percent."
+
+    # Remove temporary column.
+    table = table.drop(columns="check")
```

Comment:
For consistency with other uses that you make of `pd.DataFrame.drop`, I'd provide the value for argument `columns` as a list, i.e.

```python
table = table.drop(columns=["check"])
```